pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -217,22 +235,81 @@ loop:
 			// even on creating. We prefer most restrictive SCC in this case even if it mutates a pod.
 			allowedPod = podCopy
 			allowingProvider = provider
-			klog.V(5).Infof("pod %s (generate: %s) validated against provider %s with mutation", pod.Name, pod.GenerateName, sccName)
-			break loop
+			klog.V(5).Infof("pod %s (generate: %s) validated against provider %s with mutation", pod.Name, pod.GenerateName, provider.GetSCCName())
+			continue // keep going to find the next provider to generate a good reason string
 		case apiequality.Semantic.DeepEqual(pod, podCopy):
 			// if we don't allow mutation, only use the validated pod if it didn't require any spec changes
 			allowedPod = podCopy
 			allowingProvider = provider
-			klog.V(5).Infof("pod %s (generate: %s) validated against provider %s without mutation", pod.Name, pod.GenerateName, sccName)
+			klog.V(5).Infof("pod %s (generate: %s) validated against provider %s without mutation", pod.Name, pod.GenerateName, provider.GetSCCName())
 			break loop
 		default:
-			klog.V(5).Infof("pod %s (generate: %s) validated against provider %s, but required mutation, skipping", pod.Name, pod.GenerateName, sccName)
+			klog.V(5).Infof("pod %s (generate: %s) validated against provider %s, but required mutation, skipping", pod.Name, pod.GenerateName, provider.GetSCCName())
+			failures[provider.GetSCCName()] = fmt.Sprintf("failures final validation after mutating admission")
+		}
+	}
+
+	// add audit annotations
+	if specMutationAllowed {
+		// find next provider that was not chosen
+		var nextNotChosenProvider sccmatching.SecurityContextConstraintsProvider
+		for _, provider := range providers[i+1:] {
```

Comment:
You should ensure that `i` is not the last element in the list, otherwise you might get an error trying to access out of bounds index.

Reply:
you don't: https://goplay.space/#exE5csdtnr5

Reply:
Neat, don't you love go for those bits :smile:

Reply:
don't try `providers[i+2:]` though 💣

Reply:
so it's a delayed time bomb, only :wink: